src/qrcore.cpp

```diff
@@ -3,14 +3,47 @@
 
 #define DB this->db
 
+RCoreLocked::RCoreLocked(RCore *core)
+    : core(core)
+{
+    r_th_lock_enter(core->lock);
+}
+
+RCoreLocked::RCoreLocked(RCoreLocked&& o)
+{
+    core = o.core;
```

Comment:
In theory it should but in reality it's just a pointer and there is no such thing as moving a pointer.